src/DependencyInjection/DI/test/ServiceProviderContainerTests.cs

```diff
@@ -346,6 +346,52 @@ public void ProviderScopeDisposeThrowsWhenOnlyDisposeAsyncImplemented()
                 exception.Message);
         }
 
+        [Fact]
+        public void SingletonServiceCreatedFromFactoryIsDisposedWhenContainerIsDisposed()
+        {
+            // Arrange
+            var collection = new TestServiceCollection();
+            collection.AddSingleton(_ => new TestDisposable());
+            var provider = this.CreateServiceProvider(collection);
```

Comment:
Style nit: We don't use `this.`

Reply:
I've removed `this` in https://github.com/tomakita/Extensions/commit/0672203deab9b0ac7ef9660b0ab1f4bd88dfe958.  Thanks for pointing this out.